lib/json/jws.rb

```diff
@@ -112,9 +114,10 @@ def with_jwk_support(key)
       when JSON::JWK
         key.to_key
       when JSON::JWK::Set
-        key.detect do |jwk|
-          jwk[:kid] && jwk[:kid] == kid
-        end.try(:to_key) or raise JWK::Set::KidNotFound
+        jwk = key.detect(proc { raise JWK::Set::KidNotFound }) do |jwk|
+          jwk[:kid] && jwk[:kid] == kid && jwk.respond_to?(:to_key)
+        end
```

Comment:
I'd almost argue the `or raise` was clearer than `find` / `detect` taking a second block for default execution.